<a href="https://colab.research.google.com/github/osvaldofacundo/proyectowebscraping/blob/main/webscraping_fixture_arg2022.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
#llamamos las librerias 
from bs4 import BeautifulSoup
import requests
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import re
import plotly.express as px

In [3]:
webpage= requests.get('https://ar.marca.com/claro/futbol/primera-division/fixture.html') #la página de donde sacaremos la data
soup = BeautifulSoup(webpage.content, "html.parser") #descargamos el html

In [12]:
# solo 18 fechas que son las que se jugaron y estan completas a la fecha 16/09/22
fecha1= soup.find_all(id="jornada1") # identificamos la tabla y buscams y extraemos ella del codigo html
fecha2= soup.find_all(id="jornada2")
fecha3= soup.find_all(id="jornada3")
fecha4= soup.find_all(id="jornada4")
fecha5= soup.find_all(id="jornada5")
fecha6= soup.find_all(id="jornada6")
fecha7= soup.find_all(id="jornada7")
fecha8= soup.find_all(id="jornada8")
fecha9= soup.find_all(id="jornada9")
fecha10= soup.find_all(id="jornada10")
fecha11= soup.find_all(id="jornada11")
fecha12= soup.find_all(id="jornada12")
fecha13= soup.find_all(id="jornada13")
fecha14= soup.find_all(id="jornada14")
fecha15= soup.find_all(id="jornada15")
fecha16= soup.find_all(id="jornada16")
fecha17= soup.find_all(id="jornada17")
fecha18= soup.find_all(id="jornada18")


Transformamos a texto cada porción de el codigo html correspondiente a cada tabla

In [13]:
fecha1= str(fecha1) # transformamos a str la tabla
fecha2= str(fecha2)
fecha3= str(fecha3)
fecha4= str(fecha4)
fecha5= str(fecha5)
fecha6= str(fecha16)
fecha7= str(fecha7)
fecha8= str(fecha8)
fecha9= str(fecha9)
fecha10= str(fecha10)
fecha11= str(fecha11)
fecha12= str(fecha12)
fecha13= str(fecha13)
fecha14= str(fecha14)
fecha15= str(fecha15)
fecha16= str(fecha16)
fecha17= str(fecha17)
fecha18= str(fecha18)

In [14]:
fecha1= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha1) #cambiamos a mayusculas
fecha2= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha2)
fecha3= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha3)
fecha4= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha4)
fecha5= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha5)
fecha6= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha6)
fecha7= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha7)
fecha8= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha8)
fecha9= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha9)
fecha10= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha10)
fecha11= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha11)
fecha12= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha12)
fecha13= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha13)
fecha14= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha14)
fecha15= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha15)
fecha16= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha16)
fecha17= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha17)
fecha18= re.sub(r'<.*?>', lambda g: g.group(0).upper(), fecha18)


Pasamos cada codigo a un dataframe correspondiente a cada fecha

In [19]:
dfecha1 = pd.read_html(fecha1)[0]
dfecha2= pd.read_html(fecha2)[0]
dfecha3= pd.read_html(fecha3)[0]
dfecha4= pd.read_html(fecha4)[0]
dfecha5= pd.read_html(fecha5)[0]
dfecha6= pd.read_html(fecha6)[0]
dfecha7= pd.read_html(fecha7)[0]
dfecha8= pd.read_html(fecha8)[0]
dfecha9= pd.read_html(fecha9)[0]
dfecha10= pd.read_html(fecha10)[0]
dfecha11= pd.read_html(fecha11)[0]
dfecha12= pd.read_html(fecha12)[0]
dfecha13= pd.read_html(fecha13)[0]
dfecha14= pd.read_html(fecha14)[0]
dfecha15= pd.read_html(fecha15)[0]
dfecha16= pd.read_html(fecha16)[0]
dfecha17= pd.read_html(fecha17)[0]
dfecha18= pd.read_html(fecha18)[0]

Creamos una nueva columna de nombre fecha - donde le asignamos el numero de jornada a la que corresponde. Así podremos identificar más adelante a que jornada/fecha del campeonato corresponde cada partido

In [22]:
dfecha1['fecha']=1
dfecha2['fecha']=2
dfecha3['fecha']=3
dfecha4['fecha']=4
dfecha5['fecha']=5
dfecha6['fecha']=6
dfecha7['fecha']=7
dfecha8['fecha']=8
dfecha9['fecha']=9
dfecha10['fecha']=10
dfecha11['fecha']=11
dfecha12['fecha']=12
dfecha13['fecha']=13
dfecha14['fecha']=14
dfecha15['fecha']=15
dfecha16['fecha']=16
dfecha17['fecha']=17
dfecha18['fecha']=18

Concatenamos todos los dataframe de fechas de manera de tener uno solo con todos los partidos jugados.

In [25]:
fixture2022 = pd.concat([dfecha1, dfecha2, dfecha3, dfecha4, dfecha5, dfecha6, dfecha7, dfecha8, dfecha9,
                             dfecha10, dfecha11, dfecha12, dfecha13, dfecha14, dfecha15,
                             dfecha16, dfecha17, dfecha18], axis=0)

In [31]:
fixture2022

Equipo local Resultado                       Equipo visitante  \
0      Barracas Central       1-1  Central Córdoba (Santiago del Estero)   
1      Atlético Tucumán       1-1                         Colón Santa Fe   
2           San Lorenzo       1-1                          Independiente   
3              Banfield       1-2                      Newell's Old Boys   
4             Patronato       1-1                        Vélez Sarsfield   
..                  ...       ...                                    ...   
9              Banfield       2-1                         Colón Santa Fe   
10   Arsenal de Sarandí       2-2                        Vélez Sarsfield   
11    Newell's Old Boys       2-0                      Gimnasia La Plata   
12         Boca Juniors       1-0                            River Plate   
13  Talleres de Córdoba       1-0                     Defensa y Justicia   

    fecha  
0       1  
1       1  
2       1  
3       1  
4       1  
..    ...  
9      18  
10     18  
11     18  
12     18  
13     18  

[252 rows x 4 columns]

252 rows y cuantro columnas

descargamos el dataframe a un archivo csv

In [32]:
fixture2022.to_csv('fixtureliga_arg_2022.csv', encoding='utf-8') # pasamos el dataframe a un archivo csv